src/loaders/stm_load.c

```diff
@@ -270,7 +270,8 @@ static int stm_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	D_(D_INFO "Stored samples: %d", mod->smp);
 
 	for (i = 0; i < mod->ins; i++) {
-		if (mod->xxs[i].len > 1) {
+		if (sfh.ins[i].volume && sfh.ins[i].length) {
```

Comment:
Ok, I just offer to reproduce original loader. Playing noise instead of silence looks like a bug in a player part of ST2. Actually tracker and player are independent parts in ST2. I think that silent empty samples will be more correct than noisy, and as I know, other trackers do just that.